rust/lance-table/src/format/manifest.rs

```diff
@@ -160,6 +167,18 @@ impl Manifest {
         self.timestamp_nanos = nanos;
     }
 
+    /// Set the `table_metadata` from a metadata HashMap
+    pub fn set_metadata(&mut self, metadata: HashMap<String, String>) {
+        self.table_metadata.extend(metadata);
+    }
+
+    /// Delete `table_metadata` keys using a Vec of metadata keys
+    pub fn delete_metadata(&mut self, metadata_keys: Vec<String>) {
```

Comment:
nit: we aren't modifying the keys so we should probably just have them be a slice
```suggestion
    pub fn delete_metadata(&mut self, metadata_keys: &[&str]) {
```

Reply:
Addressed. I've also changed `lance::dataset::Dataset.delete_table_metadata` to use same argument type - is that OK?